test/libsolidity/smtCheckerTests/inline_assembly/assembly2.sol

```diff
@@ -0,0 +1,16 @@
+pragma experimental SMTChecker;
+
+contract C {
+	function f() public pure returns (bool) {
+		bool b;
+		int x = 42;
+		assembly { b := 1 }
+		assert(x == 42); // should hold
+		assert(b); // should hold, but fails due to overapproximation
+		return b;
```

Comment:
```solidity
	function f() public pure returns (bool) {
		bool b;
		int x = 42;
		assembly { b := 1 }
		b = 5
		assert(x == 42); // should hold
		assert(b); // should hold
		return b;
```
Could you also add a test like this? And a similar one for memory / storage.

Reply:
@hrkrshnn : `Error: Type int_const 5 is not implicitly convertible to expected type bool`
Did you mean `b = true;` instead of `b = 5;`?

Reply:
Yes!

Reply:
Added.